core/shared/src/main/scala/scodec/bits/ByteVector.scala

```diff
@@ -855,6 +856,23 @@ sealed abstract class ByteVector extends BitwiseOperations[ByteVector, Long] wit
     bits.toLong(signed, ordering)
 
   /**
+    * Converts the contents of this vector to a UUID.
+    *
+    * @throws IllegalArgumentException if size is not exactly 8.
+    */
+  final def toUUID: UUID = {
+    // Sanity check
+    if (size != 16) {
+      throw new IllegalArgumentException(s"Cannot convert ByteVector of size $size to UUID; must be 16 bytes")
+    }
+    // Convert
+    val byteBuffer = toByteBuffer
```

Comment:
I went for going via a ByteBuffer for simplicity since that gives us getLong/putLong "for free". I'd be happy to change to bit-twiddling if it's preferable. (I'm guessing that anyone using ByteVector isn't **that** concerned with *absolute* performance anyway, so...)